### DISCUSSION TITLE
Sentiment Analysis - Project Discussion

### DISCUSSION THREAD

**Discussion by witch-world** [ruxailab/RUXAILAB#315] [Ideas]
Hi,
My name is Huda. I want to contribute to "Sentiment Analysis Data Extraction from Usability Tests" Project. I want to ask if there is any existing repository for this project or should i need to start this project from scratch?

**Comment by jvJUCA** [comment]
Dear @witch-world,

Sorry for the delay in responding.

You are welcome to contribute to the eye tracker development in our repository. However, currently, we don't have any repositories initiated for sentiment analysis purposes. If you're interested in starting a project in this area, we would be happy to discuss it further. Thank you for your interest in contributing to our community!

Best regards,

@jvJUCA

**Reply by shashanx52** [reply]
@jvJUCA    , I would love to contribute to this domain, as I have expertise in Python, NLP, AI/ML, and model optimization. I have also worked on various sentiment analysis projects. Given the limited time, could you suggest how I can contribute effectively? Thanks!"

**Comment by witch-world** [comment]
@jvJUCA, would you please provide details regarding the dataset for this project.

**Comment by witch-world** [comment]
Because having a comprehensive understanding of the dataset is crucial for ensuring the accuracy and effectiveness of the machine learning model. 
I look forward to your response and collaborating closely on this project.

**Reply by KarinePistili** [reply]
Hello @witch-world , we currently do not have a dataset for it. The idea behind the sentiment project is to extract the information from video and text, so we could create this dataset to be analyzed and used alongside RUXAILAB. I suggest you to research about the types of extractions that can be perfomed to get information from a video recording for example. 

If you do any proof of concepts you can share with us and we can talk about it more :)

**Reply by Hacked254** [reply]
Hi @KarinePistili i did research on this and have some few key concepts that contributors on this Discussion can talk about more:
So for the Information that can be extracted from **Video Recordings** from usability tests for purposes of Sentiment Analysis include:
Facial Expressions- to determine sentiments such as positive from happy expressions or negative from frustrations and confusion
Body Language-to determine comfort/frustration
verbal tone-volume and speed of tone can convey sentiments
**For Text**
user Comments and reviews
**For Audio Recordings**
Verbal feedback
tone and Intonations- all these when analyzed can convey sentiments

**Comment by KarinePistili** [comment]
The idea behind the sentiment project is to extract the information from video and text, so we could create this dataset to be analyzed and used alongside RUXAILAB. I suggest you to research about the types of extractions that can be perfomed to get information from a video, audio and text.

If you do any proof of concepts you can share with us and we can talk about it more in this discussion. :)

**Reply by witch-world** [reply]
@KarinePistili, Just a quick query: would you please tell me where the data(video, audio, text) for the sentiment project is currently stored?

**Reply by shradiphylleia** [reply]
@KarinePistili  running a script to extract information from various sources online is something which comes to my mind at the moment.Do let me know if you are keen on some other approach. 
For extraction:-What would you consider ideal keywords to be on the lookout for?

**Reply by KarinePistili** [reply]
We still don't have a dataset or a folder containing multiple data that are public to use. What I can say is that there is a feature on-going on RUXAILAB that has screen and video recording of the users, so I believe for your tests and proposals you can ise public resources, just having in mind that other information.

**Reply by KarinePistili** [reply]
In the case of text, I suggest you to create an unmodarated test on RUXAILAB in your environment and mock some answers and start from there.

Also to study and gather the possible options and what would suit better the use case of analysing a feedback on this specific scenario which is usability testing.

**Reply by mahir-anand** [reply]
Hi Everyone,

Thanks for starting this discussion off. Just a quick question for @KarinePistili. How important is to store the dataset of the source video, audio and text? Would it be alright if I skip that step and get the associated sentiment analysis as an API response. 

I am asking this because I have been able to do this with a personal project where I took video and text input from the user and sent that as an API request and got the associated sentiment analysis (in the form of an array) as the API response. Let me know if this sounds viable and I'd be happy to share the details.

**Reply by NileshVishwakarma982** [reply]
Hi @KarinePistili ,

Thank you for providing insights into the sentiment project. I've previously worked with text data for sentiment analysis, as demonstrated in my IMDb movie review project. Additionally, I have experience with image and video processing projects, such as hand sign recognition. You can find the IMDb movie review project at [this GitHub link](https://github.com/NileshVishwakarma982/LEARNFLOW-Artificial_Intelligence/tree/main/Task3) and the hand sign recognition project at [this GitHub link](https://github.com/NileshVishwakarma982/LEARNFLOW-Artificial_Intelligence/tree/main/Task1). I'm eager to explore the types of extractions that can be performed to gather information from videos, audios, and texts. If I conduct any proof of concepts, I'll be sure to share them with you for further discussion.

**Comment by witch-world** [comment]
Because i want to go through the data for the project

**Comment by Nop-lop** [comment]
Hello there! 

My name is Nancy, and I'm from Edinburgh, UK. I'm excited to contribute to the Sentiment Analysis projects, specifically the data extraction part since it's the source material for the sentiment analysis. After doing some research, I found out that we can use different approaches for data extraction, but at the very least, we need an OCR (Optical Character Recognition) tool like Tesseract (open source) or paid technologies like TensorFlow if we have the resources.

However, before we get into the project, I have a few questions that I hope you can help me with:

- Video Text Extraction:
For video data, should we focus on extracting text from specific moments in the audio track, or should we aim for overall text detection across all video frames? For example, would we be looking for specific on-screen elements like car license plates (similar to this article: [3 min article](https://thangasami.medium.com/tesseract-vs-keras-ocr-vs-easyocr-ec8500b9455b)) or capturing all text displayed throughout the video (like in this example: [video](https://youtu.be/EkSaIJTruTA?si=_Dz5i1WnPckAGcpv))?

- Data Format Consistency:
To streamline the data extraction process, will usability test data adhere to a consistent format across different projects? For instance, would timestamps be aligned with specific user actions, or should we anticipate variations in the data structure that our extraction methods need to adapt to?

- User Anonymization:
During data processing, how can we best ensure user anonymity? Does the organization have any preferred anonymization techniques or tools for this purpose? For instance, could we consider techniques like name redaction or voice modification (for audio)?

- Usability Test Specificity:
While we can generally assume some controlled parameters for usability tests (like video length, quality, and testing conditions), are there any exceptions to this? For example, would the product or service under test be something unique and an edge-case, like a rollercoaster ride, where these typical parameters might not apply?

Thank you for your help. I'm looking forward to working on this project with you.

**Comment by mahir-anand** [comment]
Hi Everyone, 

I'm Mahir Anand, and I'm a Computer Science student based in California. I have previous experience working with sentiment analysis models and wanted to share some insights. 

As per my current understanding of the code-base, I can see two ways of accomplishing our goal here. The first one, as suggested by the maintainers, would be to extract data from the source (video or text) and then run our analysis on it. Alternatively, we could streamline the process by bypassing data extraction altogether. In this scenario, we would directly transmit the source video/text via a REST API and receive the analysis results as an API response. 

In regards to the analysis, we can rely on already existing sentiment analysis models and use their APIs (there are some really good ones, in my opinion). Alternatively, we could experiment with customizing or fine-tuning our own model for additional customization.

Let me know your thoughts. I look forward to accomplishing this goal with you all over the summer!

**Comment by shwetabh-23** [comment]
Hi @KarinePistili @jvJUCA.  I am interested in taking up this project for GSoC. I'd still like to get some clarity on this project. It says data extraction for sentiment analysis. I believe you guys want a pipeline that can perform sentiment analysis over your usability tests audio/video/text. Is this correct or do you want someone to "extract" data from somewhere, using selenium or beautiful soup? If it's the former, I have decent experience in both video and text sentiment analysis, and if it's not then I wouldn't take it. Please let me know as soon as possible as the deadline is approaching. Also, if you have any specific requirements, please let me know.

Regards,
Shwetabh

**Reply by shwetabh-23** [reply]
also, can you add the project mentor's LinkedIn so that I can contact him directly? I really think I can work on this.

**Comment by Hacked254** [comment]
Hi @KarinePistili @jvJUCA I am Ian Duncan (hibachy-Discord)also interested in this sentiment Analysis project. Kindly inform us on the Development of the repository for this project as the deadline for Proposal submission is due already. Thank you

**Comment by QingyanMu** [comment]
@jvJUCA @KarinePistili  Hello, I'm Harrison Zhang. I have submitted a proposal for the project "Data Extraction for Sentiment Analysis from Usability Tests". My ultimate goal is to develop a lightweight, multimodal sentiment analysis tool and integrate it into the existing RUXAILAB project. This tool will combine facial expression recognition, posture recognition, speech recognition, and text sentiment analysis technologies. By fusing emotional information from these diverse data sources, it aims to comprehensively evaluate the emotional state of testers during usability tests. Coupled with screen recording, the tool can automatically pinpoint areas where the user experience of the software or interface being tested is good or lacking. I would greatly appreciate any feedback you might have. Thank you!
